identity-hub-client/src/main/java/org/eclipse/dataspaceconnector/identityhub/client/IdentityHubClientImpl.java

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.client;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.eclipse.dataspaceconnector.identityhub.models.Descriptor;
+import org.eclipse.dataspaceconnector.identityhub.models.MessageRequestObject;
+import org.eclipse.dataspaceconnector.identityhub.models.RequestObject;
+import org.eclipse.dataspaceconnector.identityhub.models.ResponseObject;
+import org.eclipse.dataspaceconnector.identityhub.models.credentials.VerifiableCredential;
+
+import java.io.IOException;
+import java.rmi.ServerException;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.eclipse.dataspaceconnector.identityhub.models.WebNodeInterfaceMethod.COLLECTIONS_QUERY;
+import static org.eclipse.dataspaceconnector.identityhub.models.WebNodeInterfaceMethod.COLLECTIONS_WRITE;
+
+public class IdentityHubClientImpl implements IdentityHubClient {
+
+    private final OkHttpClient httpClient;
+    private final ObjectMapper objectMapper;
+
+    public IdentityHubClientImpl(OkHttpClient httpClient, ObjectMapper objectMapper) {
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public Collection<VerifiableCredential> getVerifiableCredentials(String hubBaseUrl) throws IOException {
+        ResponseObject responseObject;
+        Response response = httpClient.newCall(
+                        new Request.Builder()
+                                .url(hubBaseUrl)
+                                .post(buildRequestBody(COLLECTIONS_QUERY.getName()))
+                                .build())
+                .execute();
+
+        if (response.code() != 200) throw new ServerException("Server error");
```

Comment:
do not use RMI exception

Reply:
You should not reuse exceptions from JRE outside of the domain in which they are defined, custom exceptions should be introduced instead.

See JavaDoc: `A <code>ServerException</code> is thrown as a result of a remote method invocation when a <code>RemoteException</code>` ... that would be very confusing since there is no RMI invocation